core/types/block.go

```diff
@@ -550,17 +601,17 @@ func (h *Header) SanityCheck() error {
 		if h.number[i] != nil && !h.number[i].IsUint64() {
 			return fmt.Errorf("too large block number[%d]: bitlen %d", i, h.number[i].BitLen())
 		}
-		if h.difficulty[i] != nil {
-			if diffLen := h.difficulty[i].BitLen(); diffLen > 80 {
-				return fmt.Errorf("too large block difficulty[%d]: bitlen %d", i, diffLen)
-			}
-		}
 		if h.baseFee[i] != nil {
 			if bfLen := h.baseFee[i].BitLen(); bfLen > 256 {
 				return fmt.Errorf("too large base fee: bitlen %d", bfLen)
 			}
 		}
 	}
+	if h.difficulty != nil {
```

Comment:
At this point h.difficulty should be guaranteed not nil. This used to be needed because it was a pointer to an array.

Reply:
good point, removing this check